lib/application/pages/home/collections/update/update_collection_memos.dart

```diff
@@ -1,23 +1,154 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
 import 'package:layoutr/common_layout.dart';
+import 'package:memo/application/constants/animations.dart' as anims;
 import 'package:memo/application/constants/dimensions.dart' as dimens;
 import 'package:memo/application/constants/images.dart' as images;
 import 'package:memo/application/constants/strings.dart' as strings;
+import 'package:memo/application/pages/home/collections/update/update_memo_terminal.dart';
 import 'package:memo/application/theme/theme_controller.dart';
+import 'package:memo/application/view-models/home/update_collection_memos_vm.dart';
+import 'package:memo/application/view-models/home/update_collection_vm.dart';
+import 'package:memo/application/widgets/material/asset_icon_button.dart';
+import 'package:memo/application/widgets/theme/rich_text_field.dart';
+
+class UpdateCollectionMemos extends HookConsumerWidget {
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final vm = ref.read(updateCollectionMemosVM.notifier);
+    final state = ref.watch(updateCollectionMemosVM);
+    final currentPageIndex = useState(0);
+
+    final controller = usePageController(viewportFraction: 0.95);
+    final parentVM = ref.watch(updateCollectionVM.notifier);
+    ref.listen<UpdateMemosState>(updateCollectionMemosVM, (_, state) => parentVM.updateMemos(memos: state.memos));
+
+    useEffect(() {
+      void onPageUpdate() => currentPageIndex.value = controller.page!.toInt();
```

Comment:
when can page be `null`?

Reply:
Based on the docs:

```dart
  /// There are circumstances that this [PageController] can't know the current
  /// page. Reading [page] will throw an [AssertionError] in the following cases:
  ///
  /// 1. No [PageView] is currently using this [PageController]. Once a
  /// [PageView] starts using this [PageController], the new [page]
  /// position will be derived:
  ///
  ///   * First, based on the attached [PageView]'s [BuildContext] and the
  ///     position saved at that context's [PageStorage] if [keepPage] is true.
  ///   * Second, from the [PageController]'s [initialPage].
  ///
  /// 2. More than one [PageView] using the same [PageController].
```

Basically is when the controller has no or more than one corresponding `PageView`. I believe we shouldn't worry about both cases, since this fore unwrap is being made in the controller listeners, which are triggered by interactions with the `PageView` directly.